### PR TITLE
No allocation in ghost code

### PR DESCRIPTION
**Pull request by Felalolf**
I added a ghost slice and adapted the exsting files accordingly.

I created an additional branch because I expected that I will have to adapt test cases, which fortunately was not the case.

### CODE REVIEW

File: src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala [hunk 1]
```diff
@@ -95,7 +95,12 @@ trait GhostWellDef { this: TypeInfoImpl =>
 
   private def exprGhostSeparation(expr: PExpression): Messages = expr match {
     case _: PGhostExpression => noMessages
-    case e if enclosingGhostContext(e) => noMessages
+    case e if enclosingGhostContext(e) =>
+      e match {
+        case PMake(_: PGhostSliceType, _) => noMessages
+        case _: PMake | _: PNew | PReference(_: PCompositeLit) => error(e, "Allocating memory within ghost code is forbidden")
```
Comment (jcp19) [inline src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala:101]: in principle, we could allow allocation of ghost memory like ghost slices inside ghost code right? (not that it matters for now)

**Review by jcp19 — APPROVED** [review]

File: src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala (hunk 1 above)
Comment (Felalolf) [reply src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala:101]: that is what is happening at line 100

Comment (jcp19) [reply src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala:101]: Nvm, that's true

**Review by ArquintL — APPROVED** [review]
Looks good to me. Did I understand it correctly that reading a non-ghost slice is fine in ghost code but just the allocation is problematic which is now checked?

**Comment by Felalolf** [comment]
Yes

**Pull request merged**